src/wallet/interfaces.cpp

```diff
@@ -551,31 +551,42 @@ class WalletLoaderImpl : public WalletLoader
     void setMockTime(int64_t time) override { return SetMockTime(time); }
 
     //! WalletLoader methods
-    std::unique_ptr<Wallet> createWallet(const std::string& name, const SecureString& passphrase, uint64_t wallet_creation_flags, bilingual_str& error, std::vector<bilingual_str>& warnings) override
+    util::Result<std::unique_ptr<Wallet>> createWallet(const std::string& name, const SecureString& passphrase, uint64_t wallet_creation_flags, std::vector<bilingual_str>& warnings) override
     {
-        std::shared_ptr<CWallet> wallet;
         DatabaseOptions options;
         DatabaseStatus status;
         ReadDatabaseArgs(*m_context.args, options);
         options.require_create = true;
         options.create_flags = wallet_creation_flags;
         options.create_passphrase = passphrase;
-        return MakeWallet(m_context, CreateWallet(m_context, name, true /* load_on_start */, options, status, error, warnings));
+        bilingual_str error;
+        util::Result<std::unique_ptr<Wallet>> wallet{MakeWallet(m_context, CreateWallet(m_context, name, true /* load_on_start */, options, status, error, warnings))};
+        if (!wallet) {
+            return util::Error{error};
+        }
+        return wallet;
```

Comment:
nit, could save 9 lines by replacing the above 4 lines in each of the 3 methods with identical logic in one line: `return wallet ? wallet : util::Error{error};` (feel free to ignore)

Reply:
That would be good, but applying this change will result in the error below.
I'm not sure about the reason for the error.

```
wallet/interfaces.cpp:577:25: error: call to implicitly-deleted copy constructor of 'util::Result<std::unique_ptr<Wallet>>'
        return wallet ? wallet : util::Error{error};
                        ^~~~~~
./util/result.h:38:36: note: copy constructor of 'Result<std::unique_ptr<interfaces::Wallet>>' is implicitly deleted because field 'm_variant' has a deleted copy constructor
    std::variant<bilingual_str, T> m_variant;
```

Reply:
Ah, `std::unique_ptr` is move constructible and move assignable but not copy constructible or copy assignable. So this would work: `return wallet ? std::move(wallet) : util::Error{error};`

Reply:
Thanks. Done in https://github.com/bitcoin/bitcoin/pull/25616/commits/466374f6c1a032d7540b1c9307631f56fe555fcf.

So `return wallet` moves the object and `return wallet ? wallet : util::Error{error};` copies the object. Interesting.

Reply:
It might be that return with a ternary doesn't benefit from RVO (return value optimization) by the compiler in the same way as with an if statement.

https://en.cppreference.com/w/cpp/language/copy_elision

Reply:
Interesting discussion about this:

https://stackoverflow.com/questions/48250917/should-we-write-stdmove-in-the-cases-when-rvo-can-not-be-done

https://stackoverflow.com/questions/22078029/why-does-the-ternary-operator-prevent-return-value-optimization

Reply:
Thanks, not a blocker but this is a topic I plan to go deeper into in order to improve my understanding.